### ISSUE TITLE
Flameshot Crashing Frequently

### ISSUE THREAD

**Issue by merrcury** [flameshot-org/flameshot#1194]
**Flameshot version**

+ Flameshot v0.8.5
+ Compiled with Qt 5.15.1

**Describe the bug**

+ It's been crashing a lot. I had to do hard reboot to get my system to get access back. "ESC" don't work. 
+ No selector on screen. 

**To Reproduce**

+ It crashes suddenly, No RAM Issues. 

**Expected behavior**

+ It should either give pointer or exit using "ESC". 

**System Information**

+ Fedore Workstation 33 
+ Windowig System: X11

![Flameshot](https://user-images.githubusercontent.com/35988194/102639135-9b2c7980-417e-11eb-9730-d6125d82841f.png)

**Comment by westofer** [comment]
i think this is related to #170
is your notification demon working? does `notify-send "hi" "hello"` send a notification?
if not then this could be the reason it hangs

**Comment by westofer** [comment]
to kill it i had to run `pkill flameshot` twice in something like dmenu

**Comment by mmahmoudian** [comment]
@merrcury you didn't mention the Desktop Environment you are using. Please note that not all Desktop Environments come with a notification daemon (looking at you Gnome!).
The points @westofer made are correct. Flameshot for the time being depends on having a notification daemon to inform user about successful or failed tasks. Please make sure you have a notification daemon running and please report back if the issue still persists.

**Comment by mmahmoudian** [comment]
Possible duplicate of #1179

**Comment by merrcury** [comment]
Yes, It is GNOME (version 3.38.1) [Kernel: Linux 5.9.8-200.fc33.x86_64 x86_64]
I am using this extension https://extensions.gnome.org/extension/615/appindicator-support/ for system icon tray. 
It doesn't seem to be duplicate of #1179 because this happened while starting the Flameshot. 
Yes, My notification demon is working perfectly fine. (It happened today as well.)

**Comment by todarede** [comment]
I also had problems, I decided to install through apt instead of using snap. everything working fine now.

**Comment by borgmanJeremy** [comment]
@merrcury did you also have it installed via snap like @todarede ?

**Comment by merrcury** [comment]
I have used the rpm package from the GitHub Release.

**Comment by LiterallyJohnny** [comment]
I literally cannot use Flameshot at all. I have it autostarting via my i3wm configuration, and Print is bound to `flameshot gui -p "~/Pictures/Screenshots"`. Every time I try to save a screenshot, copy to keyboard, or upload to Imgur, it closes and nothing gets saved. I found that running Flameshot in the terminal using `flameshot` and then pressing the `Print` key works though. I don't know how to fixed this. I already tried reinstalling numerous times already.

**Comment by mmahmoudian** [comment]
@LiterallyJohnny please confirm that
1. on a clean boot you have flameshot running (`pgrep flameshot`)
2. You are using Flameshot version 0.8.5
3. You have notification daemon running (`notify-send "test"` and seeing the notification)
4. You have xclip installed
5. You get the same behaviour when using AppImage (you can get it from release page)

**Comment by LiterallyJohnny** [comment]
Yes, Flameshot is indeed running on a clean boot:

```
16:10:30 johnny@johnny [~] $ pgrep flameshot
531
```

Yes, I am using Flameshot version 0.8.5:

```
16:11:12 johnny@johnny [~] $ flameshot -v
Flameshot v0.8.5
Compiled with Qt 5.15.1
```

Yes, I do have a notification daemon running (Dunst), and `notify-send "test"` does indeed work.
Yes, I do have xclip installed (Version 0.13-3).
Unfortunately, I was unable to figure out how to get the AppImage to run, though I do no believe it would yield different results.

I configured i3wm on my computer to autostart Flameshot via `exec --no-startup-id flameshot &`. I can see the icon for Flameshot on my status bar, but if I right click it and press "Config", it disappears and nothing happens.

I have set my `Print` key on my keyboard to launch Flameshot GUI with my i3wm configs using `bindsym Print exec flameshot gui -p ~/Pictures/Screenshots`. The GUI shows up, but the moment I try to save a screenshot or try to copy to keyboard, it crashes and nothing happens.

However, if I were to manually start Flameshot in the terminal with `flameshot` and then using the `Print` keybind I created in the i3wm configs, it works fine.

**Comment by LiterallyJohnny** [comment]
Alright, well I found the problem and I fixed it. When I was installing and configuring Arch Linux, I accidentally put `LANG=en_us.UTF-8` in `/etc/locale.conf` instead of `LANG=en_US.UTF-8`. All's good now.

However, I am a bit curious as to why a wrong locale setting causes Flameshot to crash. It's not a big deal, but it's just a bit weird.

To @merrcury: Double check you have the correct locale setting. It could be what is causing the crashes. Good luck.

**Comment by merrcury** [comment]
I am using `en_IN.UTF-8`. I don't think so, It should interfere with flameshot.

**Comment by LiterallyJohnny** [comment]
I dunno what to do then. I had the wrong locale set in `/etc/locale.conf` and it was definitely interfering with Flameshot. It fixed itself immediately after I rebooted the computer. No issues here now.

Are you absolutely positive running `locale` gives you `en_IN.UTF-8`? What's the output for locale -a?

**Comment by mmahmoudian** [comment]
@borgmanJeremy @ZetaoYang Do you know or have a hunch on why "bad" locale can cause this?

-------

@LiterallyJohnny I appreciate all the details and investigation. Would you please also post here the output of `locale` command. Mine is as follows:

```
LANG=en_US.UTF-8
LC_CTYPE=en_US.UTF-8
LC_NUMERIC=en_US.UTF-8
LC_TIME=en_US.UTF-8
LC_COLLATE=en_US.UTF-8
LC_MONETARY=fi_FI.UTF-8
LC_MESSAGES=en_US.UTF-8
LC_PAPER=fi_FI.UTF-8
LC_NAME=fi_FI.UTF-8
LC_ADDRESS=fi_FI.UTF-8
LC_TELEPHONE=fi_FI.UTF-8
LC_MEASUREMENT=fi_FI.UTF-8
LC_IDENTIFICATION=fi_FI.UTF-8
LC_ALL=
```

**Comment by borgmanJeremy** [comment]
@mmahmoudian It's buried pretty deep in the QT stack. Many of the QT functions have referenes that state something like "assumes the operating system has properly set the locale"

**Comment by mmahmoudian** [comment]
@borgmanJeremy such dependency on the locale sounds like a logical bug to me, but perhaps we can leverage that.We can use one of these QT functions that is super sensitive to locale and run it in a try catch, and if it breaks we can show a message to user that their locale is not setup correctly.

**Comment by borgmanJeremy** [comment]
Closing this since i just merged #1324
